### PR TITLE
Allow multiple members for Medicaid application

### PR DESCRIPTION
**Pull request by hartsick**
For single member flow, it made sense to save some user details on the application.
Now with multiple members, it made sense to move that information from the application and into the member class. To do this, we made applications (Snap and Medicaid) a polymorphic association on the Member class.

To test, create a Medicaid application and confirm you can add multiple members.

### CODE REVIEW

**Comment by hartsick** [comment]
[[Medicaid MM Flow] Intro: Add a member of your household](https://trello.com/c/dWb9ERgX/429-medicaid-mm-flow-intro-add-a-member-of-your-household)

File: .gitignore [hunk 1]
```diff
@@ -14,3 +14,4 @@
 .rspec
 mkmf.log
 Guardfile
+.idea
```
Comment (jayroh) [inline .gitignore]: oooooh, do tell. What's a dot idea?

File: app/models/member.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ class Member < ApplicationRecord
     "Other",
   ].freeze
 
-  belongs_to :snap_application
+  belongs_to :benefit_application, polymorphic: true
```
Comment (jayroh) [inline app/models/member.rb]: 👍

File: db/migrate/20171017191829_extract_members_from_medicaid_app.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    safety_assured do
+      execute <<~SQL
+        UPDATE members SET snap_application_id=benefit_application_id;
+      SQL
+
+      remove_column :members, :benefit_application_id
+      remove_column :members, :benefit_application_type
+    end
+  end
+end
```
Comment (jayroh) [inline db/migrate/20171017191829_extract_members_from_medicaid_app.rb]: Oh my - nice work!

File: .gitignore (hunk 1 above)
Comment (jessieay) [reply .gitignore]: I think this is for Christa's rubymine setup!

File: db/schema.rb [hunk 4]
```diff
@@ -128,7 +125,6 @@
     t.string "residential_zip"
     t.boolean "self_employed"
     t.integer "self_employed_monthly_income"
-    t.integer "self_employment_expenses"
```
Comment (jessieay) [inline db/schema.rb]: hmmmm @hartsick it seems like this was removed accidentally -- rebase and re-migrate?

File: .gitignore (hunk 1 above)
Comment (jayroh) [reply .gitignore]: Oooooh gotcha!

Comment (hartsick) [reply .gitignore]: Yep, that's correct! It's a common editor at CfA so figured I'd toss in

File: db/schema.rb (hunk 4 above)
Comment (hartsick) [reply db/schema.rb]: Good catch!

**Pull request merged**